2.13/tests/test/test.js

```diff
@@ -445,48 +453,51 @@ function testDeprovision(instanceId, queryStrings, isAsync){
                     if (err) return done(err);
                     var message = validateJsonSchema(res.body, provisioningDeleteResponseSchema);
                     if (message!='')
-                    done(new Error(message));
-                    else
-                    done();
+                        done(new Error(message));
+                    else {
+                        lastOperationName = res.body.operation;
+                        done();
+                    }
                 })
             });
 
             if (isAsync) {
-                // TODO: the query string should contain 'operation' for the last operation. FYI: https://github.com/openservicebrokerapi/servicebroker/blob/v2.13/spec.md#parameters.
                 describe('DEPROVISION - poll', function() {
                     this.timeout(maxDelayTimeout*1000);
                     it ('should return succeeded operation status after deprovision', function(done){
-                        pollInstanceLastOperationStatus(instanceId, done);
+                        pollInstanceLastOperationStatus(instanceId, lastOperationName, done);
                     })
                 });
             }
         });
     });
 }
 
-function pollInstanceLastOperationStatus(instanceId, done) {
+function pollInstanceLastOperationStatus(instanceId, lastOperationName, done) {
     var count = 0;
     var lastOperationState = 'in progress';
     async.whilst(
         function() {
             return lastOperationState == "in progress" && count <= config.maxPollingNum;
         },
         function(callback) {
-            count++;
-            console.log(count.toString() + 'th polling last operation...');
-            preparedRequest()
-                .get('/v2/service_instances/' + instanceId + '/last_operation')
-                .set('X-Broker-API-Version', apiVersion)
-                .auth(config.user, config.password)
-                .expect(200)
-                .expect('Content-Type', /json/)
-                .end(function(err, res){
-                    if (err) callback(err);
-                    var message = validateJsonSchema(res.body, lastOperationSchema);
-                    if (message != "") callback(new Error(message));
-                    lastOperationState = res.body.state;
-                })
             setTimeout(function() {
+                count++;
+                console.log(count.toString() + 'th polling last operation...');
+                var url = '/v2/service_instances/' + instanceId + '/last_operation';
+                if (lastOperationName) url += '?operation=' + lastOperationName;

```

Comment:
I think it would be better if you could assert if `lastOperationName == ""`

Reply:
From the [spec](https://github.com/openservicebrokerapi/servicebroker/blob/v2.13/spec.md#parameters), adding this query string or not depends on whether the last operation returns an `operation` in response body, right?

Reply:
Yes